pm/queue.go

```diff
@@ -121,6 +106,10 @@ func (q *ticketQueue) Length() int32 {
 // is sufficient to cover the face value of the ticket at the head of the queue. If the max float is sufficient, we pop
 // the ticket at the head of the queue and send it into q.redeemable which an external listener can use to receive redeemable tickets
 func (q *ticketQueue) startQueueLoop() {
+	blockNums := make(chan *big.Int, 10)
```

Comment:
While the queue is empty, this channel will fill up as new blocks are received and when it is full the sender will block when it tries to send another value into the channel.

Consider the following function from `TimeWatcher`:

```
func (tw *TimeWatcher) handleBlockNum(event *blockwatch.Event) {
	last := tw.LastSeenBlock()
	new := event.BlockHeader.Number
	if last == nil || last.Cmp(new) != 0 {
		tw.setLastSeenBlock(new)
		tw.blockSubFeed.Send(new)
	}
}
```

One of the subscribers to `tw.blockSubFeed` is created here and the `blockNums` channel is used by the subscriber to receive values. Once `blockNums` is full, `tw.blockSubFeed.Send()` will block preventing the `TimeWatcher` from returning from `handleBlockNum()`. This means that `TimeWatcher` will stop updating its cached values for the last seen block.

I think we can fix this by removing the use of the condvar `q.cond` to lock and wait until the queue is non-empty. Previously, the condvar allowed the loop to pause (instead of running continuously thereby avoiding a busy loop) until a ticket was actually appended to the queue. Now that we only pop from the queue when a new block is received, we could instead just wait for a block to be received on `blockNums` and skip the use of the condvar to wait for a ticket to be added to the queue.

```golang
for {
        // Condvar logic removed

	select {
	case err := <-sub.Err():
		glog.Error(err)
	case latestBlock := <-blockNums:
		// Same code that we have right now for popping tickets from the queue
		case <-q.quit:
			return
		}
	}
```